sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/QueryPlan.scala

```diff
@@ -289,6 +289,9 @@ abstract class QueryPlan[PlanType <: QueryPlan[PlanType]] extends TreeNode[PlanT
       case m: Map[_, _] => m
       case d: DataType => d // Avoid unpacking Structs
       case seq: Traversable[_] => seq.map(recursiveTransform)
+      case ExistenceJoin(exists: Attribute) =>
+        // `exists` must be an Attribute in ExistenceJoin
+        ExistenceJoin(transformExpression(exists).asInstanceOf[Attribute])
```

Comment:
Yes I feel the same. But after modification as suggested by @nsyca , this change can also be removed. Because I move the test case to `FilterPushdownSuite`, we don't have to touch the `RewritePredicateSubquery` rule.